profiler/src/main/java/com/splunk/opentelemetry/profiler/EventProcessingChain.java

```diff
@@ -37,19 +38,27 @@
   private final SpanContextualizer spanContextualizer;
   private final ThreadDumpProcessor threadDumpProcessor;
   private final TLABProcessor tlabProcessor;
-  private final PriorityQueue<RecordedEvent> buffer =
-      new PriorityQueue<>(Comparator.comparing(RecordedEvent::getStartTime));
+  private final List<RecordedEvent> buffer = new LinkedList<>();
```

Comment:
is there a reason why you are not using `ArrayList`?

Reply:
I don't think the size of the buffer per flush is well defined (or understood)...which is why I chose LinkedList instead. It's been my general understanding that LinkedList is a better choice when workloads aren't predictable/known. Because `buffer` is final we call `clear()` when flushing, which means that if we used ArrayList we would hold onto the underlying array and it would only grow, right?

Honestly, my intuition for this isn't super strong and if people feel that ArrayList would be a better choice I am happy to change it and we can microbenchmark it later.

Reply:
See this tweet from the author of the LinkedList https://twitter.com/joshbloch/status/583813919019573248?lang=en
If you wish you can always remove final from `buffer` and recreate it on each iteration or use https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#trimToSize--.

Reply:
Ok, I'll move to ArrayList, if nothing because it's more commonplace.